.github/workflows/e2e_ci.yml

```diff
@@ -129,13 +134,34 @@ jobs:
         # Asan in llvm 14 provided in ubuntu 22.04 is incompatible with
         # high-entropy ASLR in much newer kernels that GitHub runners are
         # using leading to random crashes: https://reviews.llvm.org/D148280
-        run: sudo sysctl vm.mmap_rnd_bits=28    
+        run: sudo sysctl vm.mmap_rnd_bits=28
 
       - name: Install multilib
         if: matrix.arch == 'amd64'
         run: |
           sudo apt install -y --no-install-recommends gcc-multilib g++-multilib
 
+      - name: Install deps
+        run: |
+          sudo apt install -y --no-install-recommends clang gcc llvm build-essential cmake
+
+      - name: Install kernel headers (actuated)
+        uses: self-actuated/get-kernel-sources@master
+        if: matrix.arch == 'arm64'
+
+      - name: Install kernel headers and gcc
+        if: matrix.arch == 'amd64'
+        run: |
+          sudo apt install -y --no-install-recommends linux-headers-$(uname -r) gcc-multilib g++-multilib
+
+        # We have no guarantees that the kernel version is the same for the
+        # different workers, so we rebuild the drivers.
+      - name: Rebuild drivers
```

Comment:
It would be great to avoid building them at all in the `build` step.
Also, i am not sure whether this would always cleanly rebuild the drivers (since it should already find them there, from the restored build).

Reply:
I'll see if the cmake setup makes me do that! :)
> Also, i am not sure whether this would always cleanly rebuild the drivers (since it should already find them there, from the restored build).
I was also thinking that but it turns out that it could happen that the worker where the build step is performed is different from the one in which the tests are run; therefore the kernel can be *slightly* different and the tests will fail :/